protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/Namer.kt

```diff
@@ -52,11 +52,14 @@ public interface Namer {
 
         override fun newEnumValueTypeName(enumTypeName: String, preferred: String, nameSet: Collection<String>): String {
             val typePrefix = splitWordsToSnakeCase(enumTypeName) + '_'
-            var name = splitWordsToSnakeCase(preferred).substringAfter(typePrefix)
-            name = name.uppercase()
+            var name = splitWordsToSnakeCase(preferred).let { snakeCase ->
+                snakeCase.substringAfter(typePrefix).takeUnless {
+                    "^\\d|\\W".toRegex().matches(it)
```

Comment:
Pull this regex out into a constant that's defined alongside the other constants at the top of this class:

```kt
private val disallowedTypeNamePattern = "^\\d|\\W".toRegex()
```

Reply:
Or maybe this would be better defined as an allowed pattern rather than a disallowed one. Based on the [definition of a Kotlin identifier](https://kotlinlang.org/spec/syntax-and-grammar.html#identifiers), something similar to this should work:

```kt
private val validKotlinIdentifierPattern = "^[\\p{Lu}\\p{Ll}\\p{Lt}\\p{Lm}\\p{Lo}_][\\p{Lu}\\p{Ll}\\p{Lt}\\p{Lm}\\p{Lo}\\p{Nd}_]*".toRegex()
```

Reply:
I went ahead with the `validKotlinIdentifierPattern` instead of the `disallowedTypeNamePattern`